src/main/resources/common-services/CDAP/5.0.0/package/scripts/master.py

```diff
@@ -88,33 +101,32 @@ def killOlderProcesses(self):
         # iterate over last output and execute ls /cdap/twill/<name>/instances and save its output
         for service in services:
             service = service.strip()
-            if (service != 'master.services'):
-                Execute("/usr/hdp/current/zookeeper-client/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " ls /twill/" + service + "/instances > /tmp/twill_" + service + "_instances", user='zookeeper')
-                pipe = sp.Popen( 'tail -n1 /tmp/twill_' + service + '_instances', shell=True, stdout=sp.PIPE, stderr=sp.PIPE )
-                if pipe.wait() != 0:
-                    print('fail in executing the command ls /twill/_' + service + '/_instances')
-                    return        
-                output = pipe.communicate()[0].strip()
-                if (not(output.startswith('[') and output.endswith(']') and output != '[]')):
-                    print('output value is equal to []')
-                    return
-                result = output[1:len(output)-1]
-                # execute get /cdap/twill/<name>/instances/<last_output and save its output
-                Execute("/usr/hdp/current/zookeeper-client/bin/zkCli.sh -server " + params.cdap_zookeeper_quorum + " get /twill/" + service + "/instances/" + result + " > /tmp/twill_" + service + "_instances_output", user='zookeeper')
-                pipe = sp.Popen( 'tail -n1 /tmp/twill_' + service + '_instances_output', shell=True, stdout=sp.PIPE, stderr=sp.PIPE )
-                if pipe.wait() != 0:
-                    print('fail in executing the command get /twill/' + service + '/instances')  
-                    return
-                output = pipe.communicate()[0].strip()
-                data = json.loads(output)
-                container_id = data['data']['containerId']
-                container_values = container_id.split('_')
-                applicationId = "application_" + container_values[2] + "_" + container_values[3] 
-                kinit_cmd = params.kinit_cmd_master
-                kill_application_cmd = format("{kinit_cmd} yarn application -kill " + applicationId)
-                Execute(kill_application_cmd, user='cdap')
-                Execute("rm -rf /tmp/twill_" + service + "_instances_output", user='zookeeper') 
-                Execute("rm -rf /tmp/twill_" + service + "_instances", user='zookeeper') 
+            if (service == 'master.services'):
+                continue
+            Execute(zkCliCommand + " ls /twill/" + service + "/instances > /tmp/twill_" + service + "_instances", user='zookeeper')
+            status, output = self.executeShellCommands('tail -n1 /tmp/twill_' + service + '_instances')
+            if status != 0:
+                Execute("rm -rf /tmp/twill_" + service + "_instances", user='zookeeper')
+                continue        
+            if (not(output.startswith('[') and output.endswith(']') and output != '[]')):
+                print('output value is equal to []')
```

Comment:
Temporary file needs to be cleaned up here as well before 'continue'

Reply:
Done